app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/WalletConnectBridgeRepository.kt

```diff
@@ -224,21 +226,29 @@ class WalletConnectBridgeRepository @Inject constructor(
         }
 
     private fun showSendTransactionNotification(peerMeta: Session.PeerMeta?, safe: Solidity.Address, data: TransactionData, referenceId: Long) {
-        val icon = peerMeta?.icons?.firstOrNull()?.let { nullOnThrow { picasso.load(it).get() } }
+        val keyguard = context.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
         val intent = ReviewTransactionActivity.createIntent(context, safe, data, referenceId)
-        val notification = localNotificationManager.builder(
-            peerMeta?.name ?: context.getString(R.string.unknown_dapp),
-            context.getString(R.string.notification_new_transaction_request),
-            PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT),
-            CHANNEL_WALLET_CONNECT_REQUESTS
-        )
-            .setSubText(safe.shortChecksumString())
-            .setLargeIcon(icon)
-            .build()
-        localNotificationManager.show(
-            referenceId.hashCode(),
-            notification
-        )
+        // Pre Android Q we will directly show the review activity if the phone is unlocked, else we show a notification
+        // TODO: Adjust check when Q is released
+        if (BuildCompat.isAtLeastQ() || Build.VERSION.SDK_INT > Build.VERSION_CODES.P || keyguard.isKeyguardLocked) {
```

Comment:
isn't ```BuildCompat.isAtLeastQ() || Build.VERSION.SDK_INT > Build.VERSION_CODES.P ``` is the same as just ```BuildCompat.isAtLeastQ()```?

Reply:
no sadly not .. `BuildCompat.isAtLeastQ()` uses some magic config vars from the device and will not work once Q is released ... for this check this would be a problem, therefore this double check.